pkgs/applications/science/electronics/kicad/default.nix

```diff
@@ -27,13 +27,19 @@ let
 
   wxGTK = if (stable)
     # wxGTK3x may default to withGtk2 = false, see #73145
-    then wxGTK30.override { withGtk2 = false; }
+    then wxGTK30.override {
+      withGtk2 = false;
+      withWebKit = true;
```

Comment:
can you clarify what `withWebKit` is for?

Reply:
No, just copied from https://github.com/NixOS/nixpkgs/blob/2165e5a4a18060822cc1fd088ce7907944acbad6/pkgs/top-level/python-packages.nix#L7525
I took that as "wxPython_4_0 requires WebKit support", so it actually should not be anything "additional".

I think wxPython_4_0 should check if the  passed-in wxGTK actually has all features enabled it needs. But I'm not sure. Or wxPython should `passthru` its version of wxGTK. I don't know which is more appropriate here. Probably both.

Reply:
i suspect the closest to wxPython checking its wxGTK(wxWidgets) dependency is "does it compile with this"
having wxPython build its specific wxWidgets and passing it through (and maybe, somehow, preferring that if both wxPython and wxGTK are passed into a package) is probably a fair solution

Reply:
So how about something like:

```wxGTK = wxPython.wxwidgets;``` 

and

```assert stdenv.isDarwin -> wxGTK.version == "special-darwin-version-of-wxwidgets```

in kicad/default.nix?

Reply:
> `wxGTK = wxPython.wxwidgets;`

i think that will pull in wxPython even if it is otherwise not required ( for example when `scriptingSupport = false;` )

Reply:
"Something like" means "properly guarded" ;)

I'm currently using:

```
assert scriptingSupport -> stable;
[…]
python = python3;
wxPython = python.pkgs.wxPython_4_0;

wxGTK = if (scriptingSupport)
    # build Kicad with same wxWidgets as wxPython, see #98634
    then wxPython.wxGTK
    else if (stable)
    # wxGTK3x may default to withGtk2 = false, see #73145
    then wxGTK30.override { withGtk2 = false; }
    # wxGTK31 currently introduces an issue with opening the python interpreter in pcbnew
    # but brings high DPI support?
    else wxGTK31.override { withGtk2 = false; };
```

(#97101 changes `wxPython.wxGTK` to `wxPython.wxwidgets`.)

I don't yet if I can do: `assert scriptingSupport -> stdenv.lib.versionAtLeast python3.pkgs.wxPython_4_0.wxGTK.version "3.0"`